cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/common/SidecarInstanceFactory.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.cassandra.spark.common;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.lang3.StringUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.cassandra.sidecar.client.SidecarInstanceImpl;
+
+public class SidecarInstanceFactory
+{
+    private SidecarInstanceFactory()
+    {
+        throw new UnsupportedOperationException("Utility class");
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SidecarInstanceFactory.class);
+
+    /**
+     * Create SidecarInstance object by parsing the input string, which is IP address or hostname and optionally includes port
+     * @param input hostname string that can optionally includes the port. If port is present, the defaultPort param is ignored.
+     * @param defaultPort port value used when the input string contains no port
+     * @return SidecarInstanceImpl
+     */
+    public static SidecarInstanceImpl createFromString(String input, int defaultPort)
+    {
+        Preconditions.checkArgument(StringUtils.isNotEmpty(input), "Unable to create sidecar instance from empty input");
```

Comment:
I think HostAndPort from guava is an alternative option to use. It is included as a transitive dependency from spark-core. I think we need to make sure the API is compatible across all spark version we support

Reply:
Not sure if I follow. What is the benefit of creating a `HostAndPort` instance here? It cannot be used to construct `SidecarInstanceImpl`

Reply:
Talked offline. We stick with the current implementation.